### ISSUE TITLE
Fix Bug, When regexFormat is defined Unique Ids are not generated

### ISSUE THREAD

**Issue by githengi** [OpenSRP/opensrp-server-web#370]
Fix Bug, When regexFormat is defined Unique Ids are not generated

**Comment by githengi** [comment]
@rehammuzzamil you could also create the PostMan tests to unit tests so that they are run as part of CI

**Comment by rehammuzzamil** [comment]
Sure @githengi , I will add unit tests for these configurations too. 
I have tested this suite on the postman, and the following are my initial investigations:
- I have figured out that adding ID-SOURCE **E** and **H** fails with 400 errors due to the invalid minLength. We have added validation of min length >= 4.
- Adding **ID-SOURCE-I** fails with 500 error due to invalid escape character found in a regex format. 
- The issues of unique id generation was due to the StackOverflow ERROR due to the empty regex provided in request payload. I have added a check to verify and match the identifier with regex format only if it is not null and not empty.
cc : @ndegwamartin

**Comment by githengi** [comment]
@rehammuzzamil you implement a validation method for both PUT and POST that checks e.g that the combination of  min size, prefix, suffix, regex make sense and its possible for generation. It should then return a BAD_REQUEST with the params that are invalid

**Comment by rehammuzzamil** [comment]
@githengi  yes it throws an IllegalArgumentException when minlength is invalid. Response says "The request contain illegal argument" with 400 error code.

**Comment by ndegwamartin** [comment]
Updated the QA/Test collection here

[OpenSRP_ID_Generator_Test_Collection2.postman_collection.json.zip](https://github.com/OpenSRP/opensrp-server-web/files/4792844/OpenSRP_ID_Generator_Test_Collection2.postman_collection.json.zip)
